src/hardware/ipxserver.cpp

```diff
@@ -212,10 +220,18 @@ bool IPX_StartServer(uint16_t portnum)
 		ipxServerSocket = SDLNet_UDP_Open(portnum);
 		if(!ipxServerSocket) return false;
 
-		for (uint16_t i = 0; i < SOCKETTABLESIZE; ++i)
-			connBuffer[i].connected = false;
+		for (auto& i : connBuffer) {
+			i.connected = false;
+		}
 
-		TIMER_AddTickHandler(&IPX_ServerLoop);
+		ipx_server_running = true;
+		ipx_server_thread  = std::thread([]() {
+                        while (ipx_server_running) {
+                                IPX_ServerLoop();
+                                std::this_thread::sleep_for(
+                                        std::chrono::milliseconds(1));
```

Comment:
This is usually a red flag when implementing threading.  Worker threads usually want to be waiting on an event rather than sleeping for an arbitrary amount of time.

I had a look `IPX_ServerLoop` and I think it would be best to use `SDLNet_CheckSockets` to implement blocking.  That way you can use a longer time-out but also be woken up right away when data arrives on the socket.

Reply:
I like this much better, was not aware of that function.